mongo/src/main/scala/it/agilelab/darwin/connector/mongo/MongoConnector.scala

```diff
@@ -1,19 +1,129 @@
+          }
+        }).toFuture(),
+        mongoConfig.timeout
+      )
+    log.debug(s"${schemas.size} loaded from MongoDB")
+    schemas.flatten
+  }
+
+  override def insert(schemas: Seq[(Long, Schema)]): Unit = {
+
+    log.debug(s"inclusion of new schemas in the collection ${mongoConfig.collection}")
+
+    schemas.foreach {
+      case (id, schema) =>
+        val document = new BsonDocument
+        document.put("_id", bson.BsonInt64(id))
+        document.put("schema", bson.BsonString(schema.toString))
+        document.put("name", bson.BsonString(schema.getName))
+        document.put("namespace", bson.BsonString(schema.getNamespace))
+
+        insertIfNotExists(mongoClient.getDatabase(mongoConfig.database).getCollection(mongoConfig.collection), document)
+    }
+  }
+
+  private def insertIfNotExists(collection: MongoCollection[Document], document: BsonDocument): Unit = {
+    try {
+      Await.result(collection.insertOne(document).toFuture(), mongoConfig.timeout)
+    } catch {
+      case ex: MongoWriteException if ex.getError.getCategory == ErrorCategory.DUPLICATE_KEY =>
+        log.info("document already present, doing nothing")
+    }
+    Unit
+  }
+
+  override def createTable(): Unit = {
+    log.debug(s"Creating collection ${mongoConfig.collection}")
+    val result = mongoClient.getDatabase(mongoConfig.database).createCollection(mongoConfig.collection)
+    result.toFuture().onComplete {
+      case Success(s) => log.info(s"collection ${mongoConfig.collection} has been correctly created. \n ${s.toString()}")
+      case Failure(t) => log.info(s"collection ${mongoConfig.collection} was not created. \n ${t.getMessage}")
+    }
+  }
+
+  override def tableExists(): Boolean = {
+    Await.result(
+      mongoClient
+        .getDatabase(mongoConfig.database)
+        .listCollectionNames()
+        .filter(x => x == mongoConfig.collection)
+        .toFuture().map(_.size),
+      mongoConfig.timeout
+    ) == 1
+  }
+
+  override def tableCreationHint(): String = {
+    s"""To create the collection from shell perform the following command:
+       |db.createCollection(${mongoConfig.collection})
+     """.stripMargin
+  }
+
+  override def findSchema(id: Long): Option[Schema] = {
+
+    val query = new BasicDBObject
+    query.put("_id", bson.BsonInt64(id))
+
+    val documents =
+      mongoClient
+        .getDatabase(mongoConfig.database)
+        .getCollection(mongoConfig.collection)
+        .find(query)
+        .toFuture()
+
+    // at max there is only one schema value (string)
+    val schemaValue: Seq[String] =
+      Await.result(documents, mongoConfig.timeout)
+        .flatMap(document =>
+          document.flatMap(field =>
+            if (field._1 == "schema") {
+              Some(field._2.asString().getValue)
+            }
+            else {
+              None
+            })
+        )
+
+    if (schemaValue.nonEmpty) {
+      Some(parser.parse(schemaValue.head))
+    } else {
+      None
+    }
```

Comment:
It looks correct but I think it's more readable this way:
```suggestion
val schemaValue: Seq[String] =
      for {
        document <- Await.result(documents, mongoConfig.timeout)
        field <- document
        if field._1 == "schema"
      } yield field._2.asString().getValue

    schemaValue.headOption.map(parser.parse)
```